gfx.cpp

```diff
@@ -7134,7 +7134,7 @@ class GfxInternal
                         Buffer &gfx_buffer = buffers_[buffer];
                         SetObjectName(gfx_buffer, buffer.name);
                         if(buffer.cpu_access == kGfxCpuAccess_None)
-                            transitionResource(gfx_buffer, D3D12_RESOURCE_STATE_PIXEL_SHADER_RESOURCE);
+                            transitionResource(gfx_buffer, kernel.isGraphics() && kernel.ps_reflection_ != nullptr ? D3D12_RESOURCE_STATE_ALL_SHADER_RESOURCE : D3D12_RESOURCE_STATE_NON_PIXEL_SHADER_RESOURCE);
```

Comment:
Could this check be wrapped up in a helper function?
Maybe like `transitionResource(gfx_buffer, GetShaderResourceState(kernel));` or something along the lines?

Reply:
done